rc/base/file.kak

```diff
@@ -0,0 +1,13 @@
+decl str mimetype
+
+hook global BufOpen .* %{ %sh{
+    if [ -z "${kak_opt_filetype}" ]; then
+        mime=$(file -b --mime-type "${kak_buffile}")
+        printf %s\\n "${mime}" | grep -q '^text/x-'
```

Comment:
So, part of my reasoning for moving that here is that mime types are not standardised for most of these (any text/x- is explicitely not standard), so whatever is used is mostly specific to the `file` command. For the few that are standardised (html, xml and mail basically), it does not seem worth it to continue to keep that mimetype option around.

In other words, as `mimetype` values are not standard, I'd rather have the `file.kak` know about most kakoune filetypes than filetype scripts know about whatever `file` generates as mimetype for their filetype. Keeping the coupling in a single optional place.